### PR TITLE
Tiny pro vue server side

### PR DESCRIPTION
**Pull request by aliceDxr**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-cli/blob/main/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/toolkits/pro/src/lib/init.ts [hunk 1]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+ */
+const getDBType = (): Promise<DBAnswers> => {
+  const question: QuestionCollection<DBAnswers> =
+  {
+    type: 'list',
+    name: 'dialect',
+    message: '请选择数据库类型：',
+    choices: [
+      { name: 'mySQL', value: 'mysql' },
+      { name: 'later', value: false },
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:79]: 要不改成暂不配置，value值是’‘

File: packages/toolkits/pro/src/lib/init.ts [hunk 2]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+
+  return inquirer.prompt(question);
+}
+/**
+ * 同步创建服务端项目文件目录、文件
+ * @answers 询问客户端问题的选择值
+ * @dbAnswers  询问服务端配置的选择值
+ */
+const createServerSync = (answers: InitAnswers, dbAnswers: DBAnswers) => {
+
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:132]: 这里干嘛留个空行，删了吧

File: packages/toolkits/pro/src/lib/init.ts [hunk 3]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+  const {
+    serverType,
+  } = answers
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:135]: 这种要换行吗，多丑

File: packages/toolkits/pro/src/lib/init.ts [hunk 4]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+
+  // 复制服务端相关目录
+  const serverFrom = utils.getTemplatePath(`server/${serverType}`);
+  const serverTo = utils.getDistPath('server');
+  fs.copyTpl(serverFrom, serverTo);
 
+  // 如果命令行配置数据库，写入config 
+  if (dbAnswers.dialect && serverType === 'eggJs') {
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:143]: serverType定义成枚举吧

File: packages/toolkits/pro/template/server/eggJs/app/database/db.config.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	"password": "123456",
+	"define": {
+		"timestamps": true,
+		"freezeTableName": true,
+		"underscored": false,
+		"createdAt": "created_at",
+		"updatedAt": "updated_at"
+	},
+	"timezone": "+08:00",
+	"database": "tiny_server_test"
```
Comment (h-ivy) [inline packages/toolkits/pro/template/server/eggJs/app/database/db.config.json:15]: 改成pro_server_test

File: packages/toolkits/pro/template/server/eggJs/app/database/db.config.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+	"host": "localhost",
+	"port": "3306",
+	"username": "root",
+	"password": "123456",
+	"define": {
+		"timestamps": true,
+		"freezeTableName": true,
+		"underscored": false,
+		"createdAt": "created_at",
+		"updatedAt": "updated_at"
```
Comment (h-ivy) [inline packages/toolkits/pro/template/server/eggJs/app/database/db.config.json:12]: 你查一下define的作用是什么，里面这么多字段，非必须的就去掉吧

File: packages/toolkits/pro/src/lib/init.ts [hunk 7]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+    const eggConfigPath = path.join(serverTo, 'app/database/db.config.json')
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:144]: serverTo？应该是serverFrom吧

File: packages/toolkits/pro/src/lib/init.ts [hunk 8]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+    const writeOrReadOptions = { encoding: 'utf8' } as const;
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:145]: 这里为啥有个as const？拷贝的代码吗？as const的作用是什么

File: packages/toolkits/pro/src/lib/init.ts [hunk 9]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+    const eggConfig = JSON.parse(fs.readFileSync(eggConfigPath, writeOrReadOptions))
+    console.log(eggConfig, 'eggConfig')
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:147]: log去掉

File: packages/toolkits/pro/src/lib/init.ts [hunk 10]
```diff
@@ -63,12 +63,102 @@ const getInitAnswers = (): Promise<InitAnswers> => {
...
+  // 复制服务端相关目录
+  const serverFrom = utils.getTemplatePath(`server/${serverType}`);
+  const serverTo = utils.getDistPath('server');
+  fs.copyTpl(serverFrom, serverTo);
 
+  // 如果命令行配置数据库，写入config 
+  if (dbAnswers.dialect && serverType === 'eggJs') {
+    const eggConfigPath = path.join(serverTo, 'app/database/db.config.json')
+    const writeOrReadOptions = { encoding: 'utf8' } as const;
+    const eggConfig = JSON.parse(fs.readFileSync(eggConfigPath, writeOrReadOptions))
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts:146]: 从146行到155行用try catch包一下，不管是readFileSync、writeFileSync还是json.parse和json.stringfy都有可能失败

File: packages/toolkits/pro/template/server/eggJs/app/database/db.config.json [hunk 11]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (h-ivy) [inline packages/toolkits/pro/template/server/eggJs/app/database/db.config.json:1]: 这个用json处理不是不行，你看看ejs，我早上是想让你定义一个config.default.js.ejs模板，通过模板去生成config.default.js的内容

File: packages/toolkits/pro/template/server/eggJs/app/database/db.config.json (hunk 5 above)
Comment (aliceDxr) [reply packages/toolkits/pro/template/server/eggJs/app/database/db.config.json:15]: 统一修改为tiny_pro_server

**Pull request closed without merging**